### PR TITLE
Update go.yml

### PR DESCRIPTION
**Pull request by gmelodie**
One of the tests (and probably more in the future) needs the `ESTUARY_TOKEN` to be set. This generates and exports it.

### CODE REVIEW

**Review by alvin-reyes — COMMENTED** [review]
I'm thinking we can use the `env` to set the `ESTUARY_TOKEN`

```
jobs:
       - name: Test
          run: .....
          env:
            ESTUARY_TOKEN: $ESTUARY_TOKEN
```

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -32,5 +32,8 @@ jobs:
     - name: Build
       run: make clean all
 
+    - name: Setup
+      run: echo "ESTUARY_TOKEN=$(./estuary setup | grep Token | cut -d ' ' -f 3)" >> $GITHUB_ENV
+
     - name: Test
-      run: go test -v ./...
+      run: ESTUARY_TOKEN=$ESTUARY_TOKEN go test -v ./...
```
Comment (marshall) [inline .github/workflows/go.yml:39]: hrm, `ESTUARY_TOKEN` should already be in the environment, i.e. adding `ESTUARY_TOKEN=$ESTUARY_TOKEN` before `go test` seems redundant.

if I'm reading the [docs](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-6) correctly, you can probably just omit the this altogether

**Review by marshall — APPROVED** [review]
looks good! check my comment about the `ESTUARY_TOKEN` env var, and see if it makes sense, but otherwise this looks good to merge

File: .github/workflows/go.yml (hunk 1 above)
Comment (gmelodie) [reply .github/workflows/go.yml:39]: Correct! Done and tested [here](https://github.com/gmelodie/estuary/actions/runs/1876540621)

**Comment by gmelodie** [comment]
> I'm thinking we can use the env to set the ESTUARY_TOKEN

Absolutely! Although just leaving it out would work, I prefer this as it's more verbose and obvious.

**Pull request merged**